File: Solutions/101- Symmetric Tree/Solution.py
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        queue = []

        queue.append(root.left)
        queue.append(root.right)

        while queue:
            tempRight = queue.pop()
            tempLeft = queue.pop()

            if tempLeft is None and tempRight is None:
                continue

            if tempRight is None and tempLeft or tempLeft is None and tempRight or tempLeft.val != tempRight.val:
                return False

            queue.append(tempLeft.left)
            queue.append(tempRight.right)
            queue.append(tempLeft.right)
            queue.append(tempRight.left)


        return True

File: Solutions/101- Symmetric Tree/Solution.java
import java.util.ArrayList;
import java.util.Stack;

class Solution {
    public boolean isSymmetric(TreeNode root) {
        return isSymmetric(root, root);
    }

    private boolean isSymmetric(TreeNode root1, TreeNode root2) {
        if (root1 == null && root2 == null)
            return true;

        if (root1 == null || root2 == null || root1.val != root2.val)
            return false;

        return isSymmetric(root1.left, root2.right) && isSymmetric(root1.right, root2.left);
    }
}